src/widgetastic_patternfly4/ouia.py

```diff
@@ -0,0 +1,45 @@
+import importlib
+import sys
+
+from widgetastic.utils import ParametrizedLocator
+
+
+class OUIAMixin:
+
+    ROOT = ParametrizedLocator(
+        './/*[@data-ouia-component-type="PF4/{@component_type}" and '
+        '@data-ouia-component-id="{@component_id}"]'
+    )
+
+    def __init__(self, component_type, component_id):
+        self.component_type = component_type
+        self.component_id = component_id
+
+    @property
+    def is_safe(self):
+        return "true" in self.browser.get_attribute("data-ouia-safe", self)
+
+
+class ImportHack:
+    def __getattr__(self, name):
+        if name.endswith("OUIA"):
+            return self.generate_ouia_compat_widget(name)
```

Comment:
we absolutely have to cache the generated values else imports in different modules will be different type instances
i propose a `defaultdict` that invokes `generate_ouia_compat_widget`